packages/autorest.python/autorest/preprocess/__init__.py

```diff
@@ -17,124 +17,7 @@
 from .python_mappings import CADL_RESERVED_WORDS, RESERVED_WORDS, PadType
 
 from .. import YamlUpdatePlugin, YamlUpdatePluginAutorest
-from .._utils import parse_args, get_body_type_for_description, JSON_REGEXP, KNOWN_TYPES
```

Comment:
curious why we had to move the "add an overload" logic out of preprocess and into both emitter.ts and m4reformatter. what's the difference between a tsp operation that inputs a body and a swagger operation that inputs a body. I know that we can actually have emitters come from tsp but i'm still not sure why we have to move the logic out of preprocess

Reply:
In [my first design](https://github.com/Azure/autorest.python/pull/1770), it is compatible with multi content types of swagger:
- swagger + multi content types: some are handled in m4feformatter, some are handled in preprocess
- typespec + multi content types: native overload are handled in emitter/preprocess and no-native overload are handled in preprocess.
The problem is that preprocess needs to add more logic to distinguish different scenario, which is not easy to evolve in the future, especially in view of that we may develop more about overload for typespec. And we should also avoid bring breaking for existing code.
Based on above thought, I made much effort to move adding free overload logic out of preprocess into m4reformatter/emitters.

Reply:
sorry i'm still not following. it seems that our preprocesser logic. is: if we see an operation that takes a model as input, we add an overload for streamed input. This seems like the logic will be the same for both typespec and swagger. Why is this not the case?

Reply:
- `typespec without native overload` and `typespec with native overload` are all handled in emitter
- `swagger` for overload will be handled in m4refourmatter
- preprocess doesn't contain `add_overload` logic

Above are my thought. And I think we have reach an agreement:
![image](https://user-images.githubusercontent.com/70930885/227413191-e5a68349-a74d-4715-9540-858d64b9036d.png)

Reply:
I think we're still misunderstanding each other, let's talk about it during our scrum